pyannote/audio/labeling/tasks/multilabel.py

```diff
@@ -0,0 +1,306 @@
+    annotation = annotation.subset(regular_labels).rename_labels(mapping=renaming)
+
+    if derivation_type == 'union':
+        support = annotation.support()
+        derived.update(support.rename_tracks())
+    elif derivation_type == 'intersection':
+        overlap = Timeline()
+        for track1, track2 in annotation.co_iter(annotation):
+            if track1 == track2:
+                continue
+            overlap.add(track1[0] & track2[0])
+        derived = overlap.support().to_annotation(generator=cycle([meta_label]))
+
+    return derived
+
+class MultilabelGenerator(LabelingTaskGenerator):
+    """Batch generator for training a multi-class classifier on BabyTrain
+
+    Parameters
+    ----------
+    feature_extraction : `pyannote.audio.features.FeatureExtraction`
+        Feature extraction
+    protocol : `pyannote.database.Protocol`
+    subset : {'train', 'development', 'test'}
+    frame_info : `pyannote.core.SlidingWindow`, optional
+        Override `feature_extraction.sliding_window`. This is useful for
+        models that include the feature extraction step (e.g. SincNet) and
+        therefore output a lower sample rate than that of the input.
+    frame_crop : {'center', 'loose', 'strict'}, optional
+        Which mode to use when cropping labels. This is useful for models
+        that include the feature extraction step (e.g. SincNet) and
+        therefore use a different cropping mode. Defaults to 'center'.
+    duration : float, optional
+        Duration of sub-sequences. Defaults to 3.2s.
+    batch_size : int, optional
+        Batch size. Defaults to 32.
+    per_epoch : float, optional
+        Total audio duration per epoch, in days.
+        Defaults to one day (1).
+    parallel : int, optional
+        Number of prefetching background generators. Defaults to 1.
+        Each generator will prefetch enough batches to cover a whole epoch.
+        Set `parallel` to 0 to not use background generators.
+
+    Usage
+    -----
+    # precomputed features
+    >>> from pyannote.audio.features import Precomputed
+    >>> precomputed = Precomputed('/path/to/mfcc')
+
+    # instantiate batch generator
+    >>> batches =  MultilabelGenerator(precomputed)
+
+    # evaluation protocol
+    >>> from pyannote.database import get_protocol
+    >>> protocol = get_protocol('BabyTrain.SpeakerRole.JSALT')
+
+    # iterate over training set
+    >>> for batch in batches(protocol, subset='train'):
+    >>>     # batch['X'] is a (batch_size, n_samples, n_features) numpy array
+    >>>     # batch['y'] is a (batch_size, n_samples, n_tasks) numpy array
+    >>>     pass
+    """
+
+    def __init__(self, feature_extraction, protocol, labels,
+                 subset='train', frame_info=None, frame_crop=None,
+                 duration=3.2, batch_size=32, per_epoch=1, parallel=1,
+                 shuffle=True):
+
+        self.labels_spec = labels
```

Comment:
All over `pyannote-audio`, I make sure to use the same name for the parameter and the attribute.
It should therefore be something like `self.labels = labels` but I agree that `labels_spec` is a better name.
So, can you use `labels_spec` everywhere?